cmd/snap-seccomp/main.go

```diff
@@ -789,28 +795,128 @@ func complainAction() seccomp.ScmpAction {
 	return seccomp.ActAllow
 }
 
+var osCreateTemp = os.CreateTemp
+
+func exportBPF(fout *os.File, filter *seccomp.ScmpFilter) (bpfLen int64, err error) {
+	errPrefixFmt := "cannot export bpf filter: %w"
+
+	oldPos, err := fout.Seek(0, os.SEEK_CUR)
+	if err != nil {
+		return 0, fmt.Errorf(errPrefixFmt, err)
+	}
+	if err := filter.ExportBPF(fout); err != nil {
+		return 0, fmt.Errorf(errPrefixFmt, err)
+	}
+	nowPos, err := fout.Seek(0, os.SEEK_CUR)
+	if err != nil {
+		return 0, fmt.Errorf(errPrefixFmt, err)
+	}
+
+	return nowPos - oldPos, nil
+}
+
+// keep in sync with seccomp-support.c
```

Comment:
this probably should say a bit more about now possibly two section existing one of allow and deny, with deny taking precedence?

Reply:
Done in  [69400f9022f9dd3a10ac120deebc40304d864a10](https://github.com/snapcore/snapd/pull/13443/commits/69400f9022f9dd3a10ac120deebc40304d864a10)